server/app/controllers/admin/AdminApplicationController.java

```diff
@@ -143,15 +150,34 @@ public Result downloadSingleVersion(Http.Request request, long programId)
     }
   }
 
+  private Optional<Instant> parseDateFromQuery(Http.Request request, String queryParam) {
+    return request
+        .queryString(queryParam)
+        .filter(s -> !s.isBlank())
+        .map(
+            s -> {
+              try {
+                return dateConverter.parseIso8601DateToStartOfDateInstant(s);
+              } catch (DateTimeParseException e) {
+                throw new BadRequestException("Malformed query param: " + queryParam);
+              }
+            });
+  }
+
   /**
    * Download a CSV file containing demographics information of the current live version.
    * Demographics information is collected from answers to a collection of questions specially
    * marked by CiviForm admins.
    */
   @Secure(authorizers = Authorizers.Labels.CIVIFORM_ADMIN)
-  public Result downloadDemographics() {
+  public Result downloadDemographics(Http.Request request) {
+    TimeFilter submitTimeFilter =
+        TimeFilter.builder()
+            .setFromTime(parseDateFromQuery(request, Filters.FROM_DATE_QUERY_PARAM))
+            .setToTime(parseDateFromQuery(request, Filters.TO_DATE_QUERY_PARAM))
```

Comment:
suggestion (blocking): making the query params part of the method signature rather than pulling them off the `request` object makes it a little easier to see what the HTTP interface is. see comment on the `routes` file and [example usage](https://github.com/seattle-uat/civiform/blob/main/server/app/controllers/api/ProgramApplicationsApiController.java#L58-L64)